File: apps/services/documents/Documents.Application/Services/AuditService.cs
using System.Text.Json;
using Documents.Domain.Entities;
using Documents.Domain.Interfaces;
using Documents.Application.Models;
using Microsoft.Extensions.Logging;

namespace Documents.Application.Services;

public sealed class AuditService
{
    private readonly IAuditRepository _repo;
    private readonly ILogger<AuditService> _log;

    public AuditService(IAuditRepository repo, ILogger<AuditService> log)
    {
        _repo = repo;
        _log  = log;
    }

    public async Task LogAsync(
        string         eventName,
        RequestContext ctx,
        Guid?          documentId = null,
        string         outcome    = "SUCCESS",
        object?        detail     = null)
    {
        try
        {
            var audit = new DocumentAudit
            {
                Id            = Guid.NewGuid(),
                TenantId      = ctx.Principal.TenantId,
                DocumentId    = documentId,
                Event         = eventName,
                ActorId       = ctx.Principal.UserId,
                ActorEmail    = ctx.Principal.Email,
                Outcome       = outcome,
                IpAddress     = ctx.IpAddress,
                UserAgent     = ctx.UserAgent,
                CorrelationId = ctx.CorrelationId,
                Detail        = detail is null ? null : JsonSerializer.Serialize(detail),
                OccurredAt    = DateTime.UtcNow,
            };

            await _repo.InsertAsync(audit);
        }
        catch (Exception ex)
        {
            // Audit failures are non-fatal — log and continue
            _log.LogError(ex, "Audit insert failed for event {Event} on document {DocId}", eventName, documentId);
        }
    }
}

File: _archived/documents-nodejs/src/application/audit-service.ts
import { AuditRepository }   from '@/infrastructure/database/audit-repository';
import type { CreateAuditInput } from '@/domain/entities/document-audit';

/**
 * AuditService — thin wrapper around AuditRepository.
 * Isolates callers from the persistence detail.
 * Audit failures are non-fatal (repository swallows and logs).
 */
export class AuditService {
  async log(input: CreateAuditInput): Promise<void> {
    await AuditRepository.insert(input);
  }

  async getDocumentHistory(documentId: string, tenantId: string) {
    return AuditRepository.listForDocument(documentId, tenantId);
  }
}

export const auditService = new AuditService();
